Issue with docker-compose

I'm trying to use the Docker image by creating a docker-compose file as outlined in [this post](https://quartz.jzhao.xyz/notes/docker/). However, when I do `docker-compose up`, it throws the following error and exits:
```
quartz-hugo  | ERROR 2023/03/28 14:30:50 Failed to read Git log: fatal: detected dubious ownership in repository at '/quartz'
quartz-hugo  | To add an exception for this directory, call:
quartz-hugo  |
quartz-hugo  |  git config --global --add safe.directory /quartz
quartz-hugo  | Error: Error building site: logged 1 error(s)
quartz-hugo  | Built in 954 ms
quartz-hugo  | make: *** [Makefile:21: serve] Error 255
```
Now I've searched this and I think there is a conflict between the ownership of the mounted folder and the parent of the folder. How do I solve this problem, though?

**Context**
I'm on a PC running WSL on WIndows 11 with Docker Desktop running on the WSL2 backend.

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,6 +1,7 @@
 FROM alpine:3.16
 
 RUN apk add --no-cache go hugo git make perl
+RUN git config --global --add safe.directory '/quartz'
 RUN go install github.com/jackyzha0/hugo-obsidian@latest
 ENV PATH="/root/go/bin:$PATH"
 RUN git clone https://github.com/jackyzha0/quartz.git /quartz

